modules/rubiconAnalyticsAdapter.js

```diff
@@ -172,6 +183,13 @@ function sendMessage(auctionId, bidWonId) {
       return adUnits;
     }, {});
 
+    // We need to mark each cached bid response with its appropriate rubicon site-zone id
+    // This allows the bidWon events to have these params even in the case of a delayed render
+    Object.keys(auctionCache.bids).forEach(function (bidId) {
+      let adCode = auctionCache.bids[bidId].adUnit.adUnitCode;
+      Object.assign(auctionCache.bids[bidId], _pick(adUnitMap[adCode], ['accountId', 'siteId', 'zoneId']));
```

Comment:
shouldn't `account/site/zone` already have been added to the every bid?

Reply:
Yes, but there is a scenario that if the main message is sent, then a bidWon event comes later, the bidWon data set needs to be able to tell which site + zone it  needs.

In the normal flow it does because `adUnitMap` is defined and has all the information we need.

However, if a bidWon was delayed a few seconds and does not fire in time to be apart of the initial analytics message, this adUnitMap is `undefined` and the only other object it has access to is the `auctionCache.bids` which is why I set each bid to have its own `site Zone` associated with its adUnit.

Reply:
Obviously there are a few different ways we can do this, this is just what I came up with.

Willing to change if you want!